### PR TITLE
Feature/030 zenedely fixes

### PR DESCRIPTION
**Pull request by samchrisinger**
## Purpose

Zendeley users went through the OAuth dance but were still unable to see their content from the node settings panel. This PR fixes this.

## Changes

At no point was the AddonUserSettings instance getting created, which prevented the ExternalAccount from being listed as one of the available ones. This broke the chain of ExternalAccount events and never linked the external account to the user settings instance.

## Side Effects

@lyndsysimon should review this change before merge

### CODE REVIEW

File: website/oauth/views.py [hunk 1]
```diff
@@ -49,5 +49,6 @@ def oauth_callback(service_name, auth):
     if provider.account not in user.external_accounts:
         user.external_accounts.append(provider.account)
         user.save()
+    user.get_or_add_addon(service_name)
```
Comment (sloria) [inline website/oauth/views.py]: Do you need to do `user.save()` here?

Comment (lyndsysimon) [inline website/oauth/views.py]: I understand why you added this, but this isn't the right place for it. The OAuth infrastructure should remain independent of addons.

**Comment by samchrisinger** [comment]
@lyndsysimon, this was reimplemented using signals. The OAuthAddonUserSettingsBase binds to this signal and instantiates the user settings.

**Comment by lyndsysimon** [comment]
:+1: Love the new implementation using signals. Go.

File: website/oauth/views.py [hunk 2]
```diff
@@ -50,4 +50,8 @@ def oauth_callback(service_name, auth):
         user.external_accounts.append(provider.account)
         user.save()
 
+    oauth_complete.send({
```
Comment (sloria) [inline website/oauth/views.py]: The first argument should be the "sender" of the signal; in this case, that probably means the `provider`.

The subsequent argument are just additional data. So how about

```python
oauth_complete.send(provider, user)
```

This has the added benefit of enforcing the signature of listeners (e.g. an error will be raised if a listener doesn't include both the provider and user arguments)

File: website/addons/base/__init__.py [hunk 3]
```diff
@@ -527,6 +529,12 @@ def external_accounts(self):
             if x.provider == self.oauth_provider.short_name
         ]
 
+    @oauth_complete.connect
+    def oauth_complete(provider, account, user):
```
Comment (sloria) [inline website/addons/base/__init__.py]: Hm, this probably doesn't need to be a method `AddonOAuthUserSettingsBase`

File: website/addons/base/__init__.py [hunk 4]
```diff
@@ -527,6 +529,12 @@ def external_accounts(self):
...
+        if not user or not account:
+            return
+        user.get_or_add_addon(account.provider)
```
Comment (sloria) [inline website/addons/base/__init__.py]: Again, do you need to save `user`?

Comment (sloria) [inline website/addons/base/__init__.py]: Also, `user.add_addon` makes sense here, since you're not doing anything with the return value.

**Comment by sloria** [comment]
Final changes made during merge.

**Pull request merged**